lib/Sema/TypeCheckGeneric.cpp

```diff
@@ -174,6 +174,24 @@ Type CompleteGenericTypeResolver::resolveDependentMemberType(
     return DependentMemberType::get(baseTy, assocType, TC.Context);
   }
 
+  // If the nested type comes from a type alias, use either the alias's
+  // concrete type, or resolve its components down to another dependent member.
+  if (auto alias = nestedPA->getTypeAliasDecl()) {
+    if (nestedPA->isConcreteType())
+      return nestedPA->getConcreteType();
+
```

Comment:
Can you just use TC.substMemberTypeWithBase(dc->getModuleContext(), alias, baseTy, true) here?